python/python_fastapi_server/routers/edr.py

```diff
@@ -0,0 +1,876 @@
+
+def get_edr_collections():
+    return global_edr_collections
+
+def get_point_value(
+    edr_collectioninfo: dict,
+    instance: str,
+    coords: list[float],
+    parameters: list[str],
+    datetime_par: str,
+    z_par: str = None,
+    custom_dims: str=None,
+):
+    dataset=edr_collectioninfo['dataset']
+    urlrequest = (
+        f"SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPointValue&CRS=EPSG:4326"
+        f"&DATASET={dataset}&QUERY_LAYERS={','.join(parameters)}"
+        f"&X={coords[0]}&Y={coords[1]}&INFO_FORMAT=application/json")
+    if datetime_par:
+        urlrequest += f"&TIME={datetime_par}"
+
+    if instance:
+        urlrequest += f"&DIM_reference_time={instance_to_iso(instance)}"
+    if z_par:
+        if "vertical_name" in edr_collectioninfo and edr_collectioninfo["vertical_name"]!="ELEVATION":
+            urlrequest += f"&DIM_{edr_collectioninfo['vertical_name']}={z_par}"
+        else:
+            urlrequest += f"&ELEVATION={z_par}"
+    if custom_dims:
+        urlrequest += custom_dims
+
+    logger.info("URL: %s", urlrequest)
+    status, response = call_adaguc(url=urlrequest.encode("UTF-8"))
+    logger.info("status: %d", status)
+    if status == 0:
+        return response.getvalue()
+    return None
+
+
+@edrApiApp.get(
+    "/collections/{collection_name}/position",
+    response_model=Coverage,
+    response_model_exclude_none=True,
+)
+@edrApiApp.get(
+    "/collections/{collection_name}/instances/{instance}/position",
+    response_model=Coverage,
+    response_model_exclude_none=True,
+)
+async def get_collection_position(
+        collection_name: str,
+        request: Request,
+        coords: str,
+        instance:  Union[str, None],
+        datetime_par: str = Query(default=None, alias="datetime"),
+        parameter_name: str = Query(alias="parameter-name"),
+        z_par: str=Query(alias="z", default=None),
+        crs: str=Query(default=None),
+        requested_format: str=Query(default=None, alias="f")
+):
+    allowed_params = ["coords", "datetime", "parameter-name", "z", "f", "crs"]
+    custom_dims = [k for k in request.query_params if k not in allowed_params]
+    custom_dims = ""
+    if len(custom_dims)>0:
+        for custom_dim in custom_dims:
+            custom_dims += f"&DIM_{custom_dim}={request.query_params[custom_dim]}"
+    edr_collections = get_edr_collections()
+
+    parameter_names = parameter_name.split(",")
+    latlons = wkt.loads(coords)
+    logger.info("latlons:%s", latlons)
+    coord = {
+        "lat": latlons["coordinates"][1],
+        "lon": latlons["coordinates"][0]
+    }
+    resp = get_point_value(
+        edr_collections[collection_name],
+        instance,
+        [coord["lon"], coord["lat"]],
+        parameter_names,
+        datetime_par,
+        z_par,
+        custom_dims,
+    )
+    if resp:
+        dat = json.loads(resp)
+        return CovJSONResponse(covjson_from_resp(dat, edr_collections[collection_name]['vertical_name']))
```

Comment:
Replace by this:
```suggestion
        return covjson_from_resp(dat, edr_collections[collection_name]['vertical_name'])
```
and add the following to the function decorator:
`response_class=CoverageJsonResponse,`

Reply:
I had to change the return value hint of the call `to -> CovJSONResponse`

Reply:
There is a difference between response_class en response_model. See my new comment/suggestion.